File: Final Project/Medivault Spring/MediVault-Spring/MediVault/src/main/java/com/medivault/entites/Consultation.java
package com.medivault.entites;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "consultations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Consultation {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "patient_id", nullable = false)
	private Patient patient;
	
	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "doctor_id", nullable = false)
	private Doctor doctor;
	
	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	@JoinColumn(name = "hospital_id", nullable = false)
	private Hospital hospital;
	
	private LocalDateTime date;
	
	@Column(columnDefinition = "TEXT")
	private String diagnosis;
	
	@Column(length = 50)
	private String bp;
	
	@Column(length = 50)
	private String sugar;
	
	@Column(length = 50)
	private String temperature;
	
	@Column(columnDefinition = "TEXT")
	private String medicines = "[]";
	
	@Column(columnDefinition = "TEXT")
	private String pdfPath;
}

File: Final Project/MediVault_ DotNet/MediVault/MediVault.Backend/Models/Consultation.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MediVault.Backend.Models
{
    public class Consultation
    {
        public int Id { get; set; }
        
        public int PatientId { get; set; }
        public Patient? Patient { get; set; }
        
        public int DoctorId { get; set; }
        public Doctor? Doctor { get; set; }
        
        public int HospitalId { get; set; }
        public Hospital? Hospital { get; set; }
        
        public DateTime Date { get; set; } = DateTime.Now;
        public string Diagnosis { get; set; } = string.Empty;
        
        // Vitals
        public string BP { get; set; } = string.Empty;
        public string Sugar { get; set; } = string.Empty;
        public string Temperature { get; set; } = string.Empty;
        
        // JSON string for medicines: [{ "name": "...", "dosage": "...", "frequency": "..." }]
        public string Medicines { get; set; } = "[]"; 

        public string PdfPath { get; set; } = string.Empty;
    }
}
